src/output-json-alert.c

```diff
@@ -424,6 +424,13 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
     if (unlikely(js == NULL))
         return TM_ECODE_OK;
 
+    if (p->flow && (p->flowflags & (FLOW_PKT_TOCLIENT|FLOW_PKT_TOSERVER))) {
+        if (p->flowflags & FLOW_PKT_TOCLIENT)
```

Comment:
I'm confused on this one. Is direction the direction of the stream that trigger the alert ? or is the the direction of the packet that trigger the inspection of a stream that causes an alert ?

Reply:
It's the packet triggering detection. So it may differ based on IDS/IPS mode. I'll drop it for now and see if I can come up with something more consistent.